tests/test_func/bad/param_inconsistency.catala_en

```diff
@@ -0,0 +1,33 @@
+```catala
+declaration scope S:
+  input cond content boolean
+  internal f1 content decimal depends on x content integer
+  output out content decimal
+
+scope S:
+  definition f1 of x under condition cond
+    consequence equals decimal of (x * 2)
+  definition f1 of y under condition not cond
+    consequence equals y / 2
+  definition out equals f1 of 10
+```
+
+```catala-test-inline
+$ catala typecheck
+[ERROR] The arguments of this definition don't match the declaration.
```

Comment:
can you add to the message `("x" is different from "y")` ?

Reply:
It's more complicated than that because we're comparing lists, but done.
(didn't go as far as detecting permutations though :sweat_smile: )